src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/processors/VepAnnotationProcessor.java

```diff
@@ -13,58 +13,72 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.batch.item.ItemWriter;
+import org.springframework.batch.item.ItemProcessor;
 
 import uk.ac.ebi.eva.pipeline.io.VepProcess;
 import uk.ac.ebi.eva.pipeline.model.EnsemblVariant;
 import uk.ac.ebi.eva.pipeline.parameters.AnnotationParameters;
 
+import java.io.StringWriter;
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * ItemStreamWriter that takes VariantWrappers and serialize them into a {@link VepProcess}, which will be responsible
  * for annotating the variants and writing them to a file.
  */
-public class VepAnnotationFileWriter implements ItemWriter<EnsemblVariant> {
+public class VepAnnotationProcessor implements ItemProcessor<List<EnsemblVariant>, List<String>> {
 
-    private static final Logger logger = LoggerFactory.getLogger(VepAnnotationFileWriter.class);
+    private static final Logger logger = LoggerFactory.getLogger(VepAnnotationProcessor.class);
+
+    private static final boolean SKIP_COMMENTS = true;
 
     private final AnnotationParameters annotationParameters;
 
     private final Integer chunkSize;
 
     private final Long timeoutInSeconds;
 
-    public VepAnnotationFileWriter(AnnotationParameters annotationParameters, Integer chunkSize, Long timeoutInSeconds) {
+    public VepAnnotationProcessor(AnnotationParameters annotationParameters, Integer chunkSize, Long timeoutInSeconds) {
         this.annotationParameters = annotationParameters;
         this.chunkSize = chunkSize;
         this.timeoutInSeconds = timeoutInSeconds;
     }
 
     @Override
-    public void write(List<? extends EnsemblVariant> variantWrappers) throws Exception {
-        VepProcess vepProcess = new VepProcess(annotationParameters, chunkSize, timeoutInSeconds);
+    public List<String> process(List<EnsemblVariant> ensemblVariants) throws Exception {
+        StringWriter writer = new StringWriter();
+
+        VepProcess vepProcess = new VepProcess(annotationParameters, chunkSize, timeoutInSeconds, writer, SKIP_COMMENTS);
         vepProcess.open();
 
-        for (EnsemblVariant ensemblVariant : variantWrappers) {
+        for (EnsemblVariant ensemblVariant : ensemblVariants) {
             String line = getVariantInVepInputFormat(ensemblVariant);
             vepProcess.write(line.getBytes());
             vepProcess.write(System.lineSeparator().getBytes());
         }
 
-        if (variantWrappers.size() > 0) {
-            EnsemblVariant first = variantWrappers.get(0);
-            EnsemblVariant last = variantWrappers.get(variantWrappers.size() - 1);
-            logger.trace("VEP has received {} variants from {}:{} to {}:{}", variantWrappers.size(),
-                    first.getChr(), first.getStart(), last.getChr(), last.getStart());
-        }
+        logBatch(ensemblVariants);
 
         vepProcess.flush();
         vepProcess.close();
+        writer.close();
+
+        String[] lines = writer.getBuffer().toString().split("\n");
```

Comment:
Why if instead passing an StringWriter to the VepProcess, and then extracting the lines from the writer to a list, we pass a list to the VepProcess? 

Right now we are writing the lines to the writer, then to an array, and finally to a list. If the batch is big enough, doing that change could save some memory ...